wasm.md

```diff
@@ -1438,20 +1440,102 @@ Start Function
 Module Instantiation
 --------------------
 
+There is some dependencies among definitions that require that we do them in a certain order, even though they may appear in many valid orders.
+First, functions, tables, memories and globals get *allocated*.
+Then, tables, memories and globals get *instantiated* with elements, data and initialization vectors.
+However, since (currently) globals can only make use of imported globals to be instantiated, we can initialize at allocation time.
+Finally, the start function is evoked.
+Exports may appear anywhere in a function, but can only be performed after what they refer to is allocated.
+Inlined imports and exports are safe to extract as they appear.
+Imports must appear first in a module due to validation.
+Exports only depend on what they are exporting having been allocated, so can appear right after allocation.
```

Comment:
`Exports may appear anywhere in a function ...`, is this talking about `Inlined` exports? If so, it seems to contradict what the next sentence says. If not, maybe clarify what an `Inlined` export/import is.

Reply:
Sorry, that should have been "module", not "function". Does that make it clearer?